### ISSUE TITLE
Support inputs with dots

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/angular_analyzer_plugin#82]
Seeing this cheatsheet: https://angular.io/docs/dart/latest/guide/cheatsheet.html, there are three lines that I didn't know about and I know will create false errors for users:

```<div [attr.role]="myAriaRole">```
```<div [class.extra-sparkle]="isDelightful">```
```<div [style.width.px]="mySize">```

Needs discovery on exactly how these work. Are attr, style, and class the only supported cases? Is 'extra-sparkle' really supposed to have a dash or is that outdated? Can this ever be used to affect directives & components or only native dom elements?

**Comment by zoechi** [comment]
- `attr`, `style`, and `class` are the only supported cases

- `extra-sparkle` is perfectly valid if you want `<div class="extra-sparkle"` as result

- I'm sure that only the expression part is subject to validation with `attr.` and `class.`. 
Style properties and units depends if the data can be acquired (for example from `dart:html` like event names)

**Comment by MichaelRFairhurst** [comment]
Looks like we could probably do three things for first release:

* check that attributes with dots fit a regex, ie `/(attr\.{valid dart identifier regex}|class\.{valid css class regex}|style.[a-zA-Z](\.[a-z])?)/`
* typecheck the contents like normal but don't check against the component directives/element properties
* if `.startsWith('class.')` then check that the expression is of type boolean

Anything more than that should probably get its own ticket.

**Comment by TedSander** [comment]
I don't think attr should validate to a valid dart identifier. It is for html attributes for example aria values.
This is common: [attr.aria-label]="label"

**Comment by zoechi** [comment]
In `[attr.xxx]` `xxx` only has to comply with the HTML5 attribute name rules.

An easy target would probably be to check `[style.xxx.zz]` kind of bindings. If the `.zz` part is provided, the expression has to resolve to a number. At least I can't think of a unit where the value does not have to be a number just now.

**Comment by zoechi** [comment]
https://html.spec.whatwg.org/multipage/syntax.html#before-attribute-name-state

**Comment by MichaelRFairhurst** [comment]
Merged #88
